channeldb/graph.go

```diff
@@ -2303,33 +2346,31 @@ func nodeTraversal(tx *bbolt.Tx, nodePub []byte, db *DB,
 			// the node at the other end of the channel and both
 			// edge policies.
 			chanID := nodeEdge[33:]
-			edgeInfo, err := fetchChanEdgeInfo(edgeIndex, chanID)
-			if err != nil {
-				return err
-			}
-			edgeInfo.db = db
 
-			outgoingPolicy, err := fetchChanEdgePolicy(
-				edges, chanID, nodePub, nodes,
-			)
+			channel, err := fetchChannel(chanID)
 			if err != nil {
 				return err
 			}
 
-			otherNode, err := edgeInfo.OtherNodeKeyBytes(nodePub)
-			if err != nil {
-				return err
+			var incomingPolicy, outgoingPolicy *ChannelEdgePolicy
+			if channel.Policy1 != nil && channel.Policy1.Node != nil {
```

Comment:
This used to happen because the cache was populated with an incomplete policy struct. While refactoring the common function with `ChanUpdatesInHorizon` I found the cause and made so that `.Node` is always present. 

Also, the code now uses `channel.Info.NodeKey1Bytes` and is much simpler!